pandas/util/testing.py

```diff
@@ -2078,6 +2079,48 @@ def inner(*args, **kwargs):
     return wrapper
 
 
+def test_parallel_diff_args(num_threads=2, args_list=None, kwargs_list=None):
```

Comment:
Something like this is often needed for testing functions that alter the state of the input, such as some sorting algos and operations like `drop_duplicates`.  This could be combined with `test_parallel` to reduce code duplication, but having it in a separate function saves some if-else that might affect timing.

As a side note I think `import threading` should be done somewhere else, because a second function call would otherwise incur a lower cost, although I haven't really observed an appreciable difference yet.